File: examples/python/01_quickstart.py
"""AGR Quickstart — protect any AI agent tool call in 3 lines."""

import os

from agr import AGRClient

api_key = os.environ.get("AGR_API_KEY", "agr_sk_YOUR_KEY_HERE")
agr = AGRClient(api_key=api_key, base_url=os.environ.get("AGR_BASE_URL", "http://localhost:8000"))

result = agr.evaluate(
    agent="support-agent",
    action="read_ticket_status",
    resource="ticket-12345",
    context={"department": "support"},
)

print(f"Decision:   {result.decision}")
print(f"Risk Score: {result.risk_score}")
print(f"Risk Level: {result.risk_level}")

if result.allowed:
    print("✅ Agent may execute this action")
elif result.requires_approval:
    print(f"⏳ Approval required (id: {result.approval_id})")
    approved = agr.wait_for_approval(result.approval_id, timeout=120)
    print("✅ Approved!" if approved else "❌ Rejected or timed out")
else:
    print(f"🚫 Blocked: {result.reason}")

File: examples/node/01_quickstart.ts
import { AGRClient } from '../../packages/agr-sdk-ts/src/index.js'

const agr = new AGRClient({
  apiKey: process.env['AGR_API_KEY'] || 'agr_sk_YOUR_KEY_HERE',
  baseUrl: process.env['AGR_BASE_URL'] || 'http://localhost:8000',
})

async function main() {
  const result = await agr.evaluate('ts-agent', 'read_ticket_status', 'ticket-42', {
    department: 'engineering',
  })

  console.log(`Decision:   ${result.decision}`)
  console.log(`Risk Score: ${result.riskScore}`)
  console.log(`Allowed:    ${result.allowed}`)

  if (result.allowed) {
    console.log('✅ Agent may execute')
  } else if (result.requiresApproval) {
    console.log(`⏳ Approval required: ${result.approvalId}`)
    const approved = await agr.waitForApproval(result.approvalId!, {
      pollInterval: 2000,
      timeout: 60000,
    })
    console.log(approved ? '✅ Approved' : '❌ Rejected or timed out')
  } else {
    console.log(`🚫 Denied: ${result.reason}`)
  }
}

main().catch(console.error)
